VCU/Phantom/tasks/source/task_throttle.c

```diff
@@ -75,164 +70,136 @@ extern volatile uint8_t initializationOccured;
+                gioSetBit(BRAKE_LIGHT_PORT, BRAKE_LIGHT_PIN, BRAKE_LIGHT_ON);
 
-                // Print out DAC output
-                NumberOfChars = ltoa(throttle,(char *)command);
+                // update brake light enable in the vcu data structure
+                VCUDataPtr->DigitalOut.BRAKE_LIGHT_ENABLE = 1;
 
-                // printing debug:
-    //            sciSend(PC_UART, NumberOfChars, command);
-    //            UARTSend(PC_UART, "\r\n");
+                // update brake light state
+                previous_brake_light_state = 1;
             }
-            else
+            else if (previous_brake_light_state == 1 &&
+                    BSE_sensor_sum < BRAKING_THRESHOLD - HYSTERESIS)
             {
-                // send 0 to DAC
-                MCP48FV_Set_Value(0);
-                THROTTLE_AVAILABLE = false;
+                // turn off brake lights
+                gioSetBit(BRAKE_LIGHT_PORT, BRAKE_LIGHT_PIN, BRAKE_LIGHT_OFF);
+
+                // update brake light enable in the vcu data structure
+                VCUDataPtr->DigitalOut.BRAKE_LIGHT_ENABLE = 0;
+
+                // update brake light state
+                previous_brake_light_state = 0;
             }
 
+            /*********************************************************************************
+              Check if APPS1 and APPS2 are within 10% of each other
+             *********************************************************************************/
+
+            VCUDataPtr->DigitalVal.APPS_SEVERE_10DIFF_FAULT = check_10PercentAPPS_Fault();
+
+
+            /*********************************************************************************
+              Check if brakes are pressed and accelerator pedal
+              is pressed greater than or equal to 25%
+             *********************************************************************************/
+
+            VCUDataPtr->DigitalVal.BSE_APPS_MINOR_SIMULTANEOUS_FAULT = check_Brake_Plausibility_Fault();
 
-            // for timing:
-            gioSetBit(hetPORT1, 5, 0);
-        }else{
-            vTaskDelayUntil(&xLastWakeTime, THROTTLE_TASK_PERIOD_MS);
+            // Give vcu key back
+            xSemaphoreGive(vcuKey);
         }
-    }
 
 
+        // debugging - jaypacamarra
+        // manually setting state to RUNNING and setting THROTTLE_AVAILABLE to true to test DAC - jaypacamarra
+//        state = RUNNING;
+//        THROTTLE_AVAILABLE = true;
+
+        /*********************************************************************************
+          Set Throttle
+         *********************************************************************************/
+        if (state == RUNNING && THROTTLE_AVAILABLE)
+        {
+            // update throttle percentage in vcu data structure
+            apps_percent_avg = (Percent_APPS1_Pressed + Percent_APPS2_Pressed) / 2;
+            VCUDataPtr->AnalogOut.throttle_percentage.value = apps_percent_avg;
```

Comment:
You returned the vcuKey before this statement. So, you should retake the vcuKey before you make any changes to the VCU Data structure.

Reply:
Alright I made a commit that retakes the vcu key back there and then returns the vcu key right afterward.